### PR TITLE
Eliminate ops for parameterized type constructors

### PR DESCRIPTION
**Pull request by ajreynol**
We now preserve types when rewriting. This means that we no longer need to use operators that store the type of terms to construct in the cases of bags, sets, and sequences.

### CODE REVIEW

File: src/theory/bags/bags_rewriter.h [hunk 1]
```diff
@@ -198,13 +198,13 @@ class BagsRewriter : public TheoryRewriter
 
   /**
    *  rewrites for n include:
-   *  - (bag.from_set (singleton (SetSingletonOp Int) x)) = (bag x 1)
+   *  - (bag.from_set (singleton x)) = (bag x 1)
```
Comment (mudathirmahgoub) [inline src/theory/bags/bags_rewriter.h:201]:
```suggestion
   *  - (bag.from_set (set.singleton x)) = (bag x 1)
```

File: src/theory/bags/bags_rewriter.h [hunk 2]
```diff
@@ -198,13 +198,13 @@ class BagsRewriter : public TheoryRewriter
...
    *  rewrites for n include:
-   *  - (bag.from_set (singleton (SetSingletonOp Int) x)) = (bag x 1)
+   *  - (bag.from_set (set.singleton x)) = (bag x 1)
    */
   BagsRewriteResponse rewriteFromSet(const TNode& n) const;
 
   /**
    *  rewrites for n include:
-   *  - (bag.to_set (bag x n)) = (singleton (SetSingletonOp T) x)
+   *  - (bag.to_set (bag x n)) = (singleton x)
```
Comment (mudathirmahgoub) [inline src/theory/bags/bags_rewriter.h:207]:
```suggestion
   *  - (bag.to_set (bag x n)) = (set.singleton x)
```

File: src/theory/quantifiers/sygus/sygus_grammar_cons.cpp [hunk 3]
```diff
@@ -985,13 +980,7 @@ void CegGrammarConstructor::mkSygusDefaultGrammar(
...
       std::vector<TypeNode> cargsSingleton;
       cargsSingleton.push_back(unresElemType);
-
-      // lambda x . (singleton (singleton_op T) x) where T = x.getType()
-      Node x = nm->mkBoundVar(etype);
-      Node vars = nm->mkNode(BOUND_VAR_LIST, x);
-      Node singleton = nm->mkSingleton(etype, x);
-      Node lambda = nm->mkNode(LAMBDA,vars, singleton);
-      sdts[i].addConstructor(lambda, "singleton", cargsSingleton);
+      sdts[i].addConstructor(SET_SINGLETON, cargsSingleton);
```
Comment (mudathirmahgoub) [inline src/theory/quantifiers/sygus/sygus_grammar_cons.cpp:983]: Sequences and sets are mentioned here. What about bags :)

**Review by mudathirmahgoub — APPROVED** [review]
LGTM, Thanks for the simplifying constructing singletons in sets, sequences and bags. 
This PR revealed bags were not handled in sygys_grammar_cons.h. 
Please handle before merge.

**Comment by mudathirmahgoub** [comment]
This line is missed https://github.com/cvc5/cvc5/blob/main/src/theory/strings/theory_strings_utils.cpp#L166

File: src/theory/quantifiers/sygus/sygus_grammar_cons.cpp (hunk 3 above)
Comment (ajreynol) [reply src/theory/quantifiers/sygus/sygus_grammar_cons.cpp:983]: Adding a default grammar for bags will take some thought. I will do this separately.

**Pull request merged**